app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,38 @@
+class ApplicationsController < ApplicationController
+
+  def new
+  end
+
+  def create
+    application = Application.new(application_params)
+    if application.save
+      redirect_to "/applications/#{application.id}"
+    else
+      flash[:alert] = "Please fill in all required fields"
+      # flash[:alert] = "Error: #{error_message(@application.errors)}"
+      redirect_to "/applications/new"
+      # render :new
```

Comment:
Great work handling the sad path! You can either redirect OR render the form again here. The only difference is that `render` will not make an additional HTTP request and will just render the form again. This means, the form could end up on the page again without changing the URI from the previous request